ddl/ddl_api.go

```diff
@@ -1899,7 +1899,50 @@ func BuildTableInfo(
 		tbInfo.Indices = append(tbInfo.Indices, idxInfo)
 	}
 
-	return
+	err = addIndexForForeignKey(ctx, tbInfo)
+	return tbInfo, err
+}
+
+// addIndexForForeignKey uses to auto create an index for the foreign key if the table doesn't have any index cover the
+// foreign key columns.
+func addIndexForForeignKey(ctx sessionctx.Context, tbInfo *model.TableInfo) error {
+	if len(tbInfo.ForeignKeys) == 0 {
+		return nil
+	}
+	var handleCol *model.ColumnInfo
+	if tbInfo.PKIsHandle {
+		handleCol = tbInfo.GetPkColInfo()
+	}
+	for _, fk := range tbInfo.ForeignKeys {
+		if fk.Version < 1 {
+			continue
+		}
+		if handleCol != nil && len(fk.Cols) == 1 && handleCol.Name.L == fk.Cols[0].L {
+			continue
+		}
+		if model.FindIndexByColumns(tbInfo, fk.Cols...) != nil {
+			continue
+		}
+		for _, idx := range tbInfo.Indices {
+			if idx.Name.L == fk.Name.L {
```

Comment:
If we need to check it here, do we need to check it to `create index`? It seems that we do check i in `create index`

Reply:
Here is an example:

```
>create table t1 (id int key, b int, index(b));
>create table t2 (a int, b int, index fk_1(a), foreign key (b) references t1(b));  
--ERROR: [ddl:1061]duplicate key name fk_1
```

since the `foreign key (b)` doesn't have a related index, so we need to auto-create a index which name is same with the foreign key, which is `fk_1`, then the index name will be duplicate, so need to return an error here.